### ISSUE TITLE
Cannot start container on CentOS 7.3 when namespaces are enabled.

### ISSUE THREAD

**Issue by datacarl** [moby/moby#34011]
**Description**

Using `userns-remap` on CentOS 7.3 fails with 

> nsenter: failed to unshare namespaces: Invalid argument
container_linux.go:262: starting container process caused "process_linux.go:247: running exec setns process for init caused \"exit status 34\""
docker: Error response from daemon: oci runtime error: container_linux.go:262: starting container process caused "process_linux.go:247: running exec setns process for init caused \"exit status 34\"".

**Steps to reproduce the issue:**
1. Configure `/etc/docker/daemon.json` with `userns-remap: default`.
2. Start/restart docker daemon.
3. Add `dockremap:296608:65536` to `/etc/subgid` and `/etc/subuid`.
4. Run hello world `docker run hello-world`.

**Describe the results you received:**
The image is downloaded but the container is not started. It fails with 

> > nsenter: failed to unshare namespaces: Invalid argument
container_linux.go:262: starting container process caused "process_linux.go:247: running exec setns process for init caused \"exit status 34\""
docker: Error response from daemon: oci runtime error: container_linux.go:262: starting container process caused "process_linux.go:247: running exec setns process for init caused \"exit status 34\"".

**Describe the results you expected:**
Container starts.

**Additional information you deem important (e.g. issue happens only occasionally):**

Fails regardless of `selinux` being set to `permissive/enforcing/disabled`.

A directory for the `dockremap` user is created in `/var/lib/docker` as expected.

Disabling namespaces solves the problem. Reenabling namespaces reintroduces it.

Logs from the daemon

```
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.971975091Z" level=debug msg="Calling GET /_ping"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.974963272Z" level=debug msg="Calling POST /v1.30/containers/create"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.975355971Z" level=debug msg="form data: {\"AttachStderr\":true,\"AttachStdin\":false,\"AttachStdout\":true,\"Cmd\":null,\"Domainname\":\"\",\"Entrypoint\":null,\"Env\":[],\"HostConfig\":{\"AutoRemove\":false,\"Binds\":null,\"BlkioDeviceReadBps\":null,\"BlkioDeviceReadIOps\":null,\"BlkioDeviceWriteBps\":null,\"BlkioDeviceWriteIOps\":null,\"BlkioWeight\":0,\"BlkioWeightDevice\":null,\"CapAdd\":null,\"CapDrop\":null,\"Cgroup\":\"\",\"CgroupParent\":\"\",\"ConsoleSize\":[0,0],\"ContainerIDFile\":\"\",\"CpuCount\":0,\"CpuPercent\":0,\"CpuPeriod\":0,\"CpuQuota\":0,\"CpuRealtimePeriod\":0,\"CpuRealtimeRuntime\":0,\"CpuShares\":0,\"CpusetCpus\":\"\",\"CpusetMems\":\"\",\"DeviceCgroupRules\":null,\"Devices\":[],\"DiskQuota\":0,\"Dns\":[],\"DnsOptions\":[],\"DnsSearch\":[],\"ExtraHosts\":null,\"GroupAdd\":null,\"IOMaximumBandwidth\":0,\"IOMaximumIOps\":0,\"IpcMode\":\"\",\"Isolation\":\"\",\"KernelMemory\":0,\"Links\":null,\"LogConfig\":{\"Config\":{},\"Type\":\"\"},\"Memory\":0,\"MemoryReservation\":0,\"MemorySwap\":0,\"MemorySwappiness\":-1,\"NanoCpus\":0,\"NetworkMode\":\"default\",\"OomKillDisable\":false,\"OomScoreAdj\":0,\"PidMode\":\"\",\"PidsLimit\":0,\"PortBindings\":{},\"Privileged\":false,\"PublishAllPorts\":false,\"ReadonlyRootfs\":false,\"RestartPolicy\":{\"MaximumRetryCount\":0,\"Name\":\"no\"},\"SecurityOpt\":null,\"ShmSize\":0,\"UTSMode\":\"\",\"Ulimits\":null,\"UsernsMode\":\"\",\"VolumeDriver\":\"\",\"VolumesFrom\":null},\"Hostname\":\"\",\"Image\":\"hello-world\",\"Labels\":{},\"NetworkingConfig\":{\"EndpointsConfig\":{}},\"OnBuild\":null,\"OpenStdin\":false,\"StdinOnce\":false,\"Tty\":false,\"User\":\"\",\"Volumes\":{},\"WorkingDir\":\"\"}"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.985509385Z" level=debug msg="container mounted via layerStore: /var/lib/docker/296608.296608/overlay/538ff82239be5a98e36222b9b813e99926ead89764e9972044665370f9ad187e/merged"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.991675060Z" level=debug msg="Calling POST /v1.30/containers/b3fc1439295cb350f26a8ae3594aa65775389a2d54e1b80ca5e807ddd581fd5e/attach?stderr=1&stdout=1&stream=1"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.991838589Z" level=debug msg="attach: stdout: begin"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.991907590Z" level=debug msg="attach: stderr: begin"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.992783996Z" level=debug msg="Calling POST /v1.30/containers/b3fc1439295cb350f26a8ae3594aa65775389a2d54e1b80ca5e807ddd581fd5e/wait?condition=next-exit"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.993502031Z" level=debug msg="Calling POST /v1.30/containers/b3fc1439295cb350f26a8ae3594aa65775389a2d54e1b80ca5e807ddd581fd5e/start"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.994180774Z" level=debug msg="container mounted via layerStore: /var/lib/docker/296608.296608/overlay/538ff82239be5a98e36222b9b813e99926ead89764e9972044665370f9ad187e/merged"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.994519771Z" level=debug msg="Assigning addresses for endpoint boring_goldstine's interface on network bridge"
Jul 07 14:17:31 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:31.994546240Z" level=debug msg="RequestAddress(LocalDefault/172.17.0.0/24, <nil>, map[])"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.014394766Z" level=debug msg="Assigning addresses for endpoint boring_goldstine's interface on network bridge"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.023932022Z" level=debug msg="Programming external connectivity on endpoint boring_goldstine (875963bad98b714f549a7ca7aba5c368be685b2bf87c7ec5ac173f6680a46723)"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.025800338Z" level=debug msg="EnableService b3fc1439295cb350f26a8ae3594aa65775389a2d54e1b80ca5e807ddd581fd5e START"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.025822729Z" level=debug msg="EnableService b3fc1439295cb350f26a8ae3594aa65775389a2d54e1b80ca5e807ddd581fd5e DONE"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.099440482Z" level=error msg="containerd: start container" error="oci runtime error: container_linux.go:262: starting container process caused \"process_linux.go:247: running exec setns process for init caused \\\"exit status 34\\\"\"\n" id=b3fc1439295cb350f26a8ae3594aa65775389a2d54e1b80ca5e807ddd581fd5e
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.100336752Z" level=error msg="Create container failed with error: oci runtime error: container_linux.go:262: starting container process caused \"process_linux.go:247: running exec setns process for init caused \\\"exit status 34\\\"\"\n"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.104186697Z" level=debug msg="attach: stdout: end"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.104212038Z" level=debug msg="attach: stderr: end"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.104542766Z" level=debug msg="Revoking external connectivity on endpoint boring_goldstine (875963bad98b714f549a7ca7aba5c368be685b2bf87c7ec5ac173f6680a46723)"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.105057000Z" level=debug msg="DeleteConntrackEntries purged ipv4:0, ipv6:0"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.125679173Z" level=debug msg="Releasing addresses for endpoint boring_goldstine's interface on network bridge"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.125705988Z" level=debug msg="ReleaseAddress(LocalDefault/172.17.0.0/24, 172.17.0.2)"
Jul 07 14:17:32 ip-10-95-5-48.eu-west-1.compute.internal dockerd[6246]: time="2017-07-07T14:17:32.128880046Z" level=error msg="Handler for POST /v1.30/containers/b3fc1439295cb350f26a8ae3594aa65775389a2d54e1b80ca5e807ddd581fd5e/start returned error: oci runtime error: container_linux.go:262: starting container process caused \"process_linux.go:247: running exec setns process for init caused \\\"exit status 34\\\"\"\n"
```

Daemon config

```
{
    "bip": "172.17.0.1/24",
    "debug": true,
    "icc": true,
    "selinux-enabled": true,
    "userns-remap": "default"
}
```

**Output of `docker version`:**

```
docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:20:36 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:21:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
docker info
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 1
Server Version: 17.06.0-ce
Storage Driver: overlay
 Backing Filesystem: xfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
 userns
Kernel Version: 3.10.0-514.16.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.518GiB
Name: ip-10-95-5-48.eu-west-1.compute.internal
ID: B5VA:GGWB:H3UG:JLUG:7OB3:JJCB:FGS5:CEKF:Y4U4:U7SG:VBQC:VY7E
Docker Root Dir: /var/lib/docker/296608.296608
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 17
 Goroutines: 24
 System Time: 2017-07-07T14:12:41.022109212Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Labels:
 com.lookback.host=54.75.205.199
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running on AWS with latest CentOS image from the marketplace.
```
cat /etc/centos-release
CentOS Linux release 7.3.1611 (Core)
```

**Comment by thaJeztah** [comment]
I suspect you don't have user names enabled in the kernel; can you try running the check-config.sh script? https://raw.githubusercontent.com/moby/moby/master/contrib/check-config.sh


If the script shows that user-namespaces are not enabled, enable the feature running;

```bash
grubby --args="user_namespace.enable=1" --update-kernel="$(grubby --default-kernel)"
```

Then reboot the host (`reboot`)

**Comment by datacarl** [comment]
Yeah that was it. Thanks a lot @thaJeztah!

**Comment by thaJeztah** [comment]
Perfect!

Wondering if there's a better way to check for this, and to produce a more useful error; I'll discuss this

**Comment by karthik101** [comment]
I'm getting the same problem, it was working when i enabled user-namespace but i have disabled selinux, found the issue is fixed in kernel 4.11 so upgraded to 4.14 and saw that issue is still present. Any workarounds?

**Comment by FelikZ** [comment]
That helped me !35806
